packages/auto-approve/src/process-checks/sample-application-repos/java-dependency.ts

```diff
@@ -0,0 +1,171 @@
+                  <artifactId>google-api-services-policytroubleshooter</artifactId>
+                  -     <version>v1-rev20210319-1.31.5</version>
+                */
+      oldVersion: new RegExp(
+        /<groupId>(?<oldDependencyNamePrefixPom>[^<]*)<\/groupId>[\s]*<artifactId>(?<oldDependencyNamePom>[^<]*)<\/artifactId>[\s]*-[\s]*<version>(?:v[0-9]-rev(?<oldRevVersionPom>[0-9]*)-(?<oldMajorRevVersionPom>[0-9]*)\.(?<oldMinorRevVersionPom>[0-9]*\.[0-9])|(?<oldMajorVersionPom>[0-9]*)\.(?<oldMinorVersionPom>[0-9]*\.[0-9]*))<\/version>[\s]*/
+      ),
+      /* This would match:
+                  <groupId>com.google.cloud</groupId>
+                  <artifactId>google-cloud-datacatalog</artifactId>
+            -     <version>1.4.1</version>
+            +     <version>1.4.2</version>
+                  or
+                   <groupId>com.google.apis</groupId>
+                   <artifactId>google-api-services-policytroubleshooter</artifactId>
+            -      <version>v1-rev20210319-1.31.5</version>
+            +      <version>v1-rev20210319-1.32.1</version>
+                */
+      newVersion: new RegExp(
+        /<groupId>(?<newDependencyNamePrefixPom>[^<]*)<\/groupId>[\s]*<artifactId>(?<newDependencyNamePom>[^<]*)<\/artifactId>[\s]*-[\s]*<version>(?:v[0-9]-rev[0-9]*-[0-9]*\.[0-9]*\.[0-9]|[[0-9]*\.[0-9]*\.[0-9]*)<\/version>[\s]*\+[\s]*<version>(v[0-9]-rev(?<newRevVersionPom>[0-9]*)-(?<newMajorRevVersionPom>[0-9]*)\.(?<newMinorRevVersionPom>[0-9]*\.[0-9])|(?<newMajorVersionPom>[0-9]*)\.(?<newMinorVersionPom>[0-9]*\.[0-9]*))<\/version>/
+      ),
+    },
+  ];
+  constructor(octokit: Octokit) {
+    super(octokit);
+  }
+
+  public async checkPR(incomingPR: PullRequest): Promise<boolean> {
+    const authorshipMatches = checkAuthor(
+      this.classRule.author,
+      incomingPR.author
+    );
+
+    const titleMatches = checkTitleOrBody(
+      incomingPR.title,
+      this.classRule.titleRegex
+    );
+
+    const fileCountMatch = checkFileCount(
+      incomingPR.fileCount,
+      this.classRule.maxFiles
+    );
+
+    const filePatternsMatch = checkFilePathsMatch(
+      incomingPR.changedFiles.map((x: {filename: string}) => x.filename),
+      this.classRule.fileNameRegex
+    );
+
+    for (const file of incomingPR.changedFiles) {
+      const fileMatch = this.fileRules?.find((x: FileRule) =>
+        x.targetFileToCheck.test(file.filename)
+      );
+
+      if (!fileMatch) {
+        return false;
+      }
+
+      const versions = getJavaVersions(
+        file,
+        fileMatch.oldVersion,
+        fileMatch.newVersion
+      );
+
+      if (!versions) {
+        return false;
+      }
+
+      const doesDependencyMatch = doesDependencyChangeMatchPRTitleV2(
+        versions,
+        // We can assert this exists since we're in the class rule that contains it
+        fileMatch.dependencyTitle!,
+        incomingPR.title
+      );
+
+      const isVersionValid = runVersioningValidation(versions);
```

Comment:
It would still be nice to do the refactor we had planned or at least make this more readable. It's hard to tell at a glance what this is validating.